### PR TITLE
Update DM allocator to use all available names

### PR DESCRIPTION
**Pull request by ConnorJC3**
**Is this a bug fix or adding new feature?**

Bug fix

**What is this PR about? / Why do we need it?**

Changes the device name allocator to use a list of all legal EBS device names.

Supersedes #1622 

**What testing is done?** 

Updated unit tests, manually tested, CI

### CODE REVIEW

**Comment by hanyuel** [comment]
Can you please add in your PR description explaining what is the bug/problem?

File: pkg/cloud/devicemanager/device_names.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package devicemanager
+
+// EC2 allowed EBS device names, discovered by trial and error
+// Notable (undoumented) restrictions include:
```
Comment (hanyuel) [inline pkg/cloud/devicemanager/device_names.go:20]: typo: ``undocumented``

File: pkg/cloud/devicemanager/allocator.go [hunk 2]
```diff
@@ -34,34 +34,19 @@ type ExistingNames map[string]string
...
-		for c2 := 'a'; c2 <= c2end; c2++ {
-			name := fmt.Sprintf("%s%s%s", prefix, string(c1), string(c2))
-			if _, found := existingNames[name]; !found {
-				return name, nil
-			}
+// GetNext returns a free device name or error when there is no free device name
+// It does this by using a list of legal EBS device names from device_names.go
+func (d *nameAllocator) GetNext(existingNames ExistingNames) (string, error) {
+	for _, name := range deviceNames {
+		if _, found := existingNames[name]; !found {
```
Comment (hanyuel) [inline pkg/cloud/devicemanager/allocator.go:48]: I know this PR doesn't change it, but why does ``ExistingNames`` need to be a map? We care about only the keys, not the values.

Comment (ConnorJC3) [reply pkg/cloud/devicemanager/allocator.go:48]: `ExistingNames` is a map because that is how it is stored and passed in `manager.go`.

**Review by torredil — APPROVED** [review]

**Pull request merged**